psycopg_c/psycopg_c/_psycopg/copy.pyx

```diff
@@ -34,10 +34,7 @@ def format_row_binary(
     else:
         pos = PyByteArray_GET_SIZE(out)
 
-    # let's start from a nice chunk
-    # (larger than most fixed size; for variable ones, oh well, we'll resize it)
-    cdef char *target = CDumper.ensure_size(
-        out, pos, sizeof(berowlen) + 20 * rowlen)
+    cdef char *target = CDumper.ensure_size(out, pos, sizeof(berowlen))
```

Comment:
As wrote above, not quite sure if this change is really needed.

On a sidenote - I still wonder, why the buffer would go out of hand here. Since `pos` indicates the last buffer write end, shouldn't oversizing things at the beginning have no effect at all? I dont know much about the code, just wondering if `pos` gets in all cases adjusted before or after dumper execution. If always done afterwards, an exception within the dumper should not report a too big buffer, as pos would still be the value from the last field adaption?

Reply:
Oh wait - since the exception will exit the code block early, the final pos-based size adjustment gets skipped and eventually the bigger current size of the buffer reported back.

Got it, sorry for the noise.